internal/resultstore/resultstore.go

```diff
@@ -56,6 +60,58 @@ func (rs *ResultStore) generatePath(p Pkg) string {
 	return path
 }
 
+func (rs *ResultStore) SaveWriteBufferZip(ctx context.Context, p Pkg, fileName string, writeBufferPaths []string) error {
+	bkt, err := blob.OpenBucket(ctx, rs.bucket)
+	if err != nil {
+		return err
+	}
+	defer bkt.Close()
+
+	uploadPath := filepath.Join(rs.generatePath(p), fileName+".zip")
+	log.Info("Uploading results",
+		"bucket", rs.bucket,
+		"path", uploadPath)
+
+	w, err := bkt.NewWriter(ctx, uploadPath, nil)
+	defer w.Close()
+	if err != nil {
+		return err
+	}
+
+	zipWriter := zip.NewWriter(w)
+	defer zipWriter.Close()
+
+	for _, filePath := range writeBufferPaths {
+		file, err := os.OpenFile(filePath, os.O_RDWR, 0666)
+		if err != nil {
+			return err
+		}
+		fileContents, err := os.ReadFile(filePath)
+		if err != nil {
+			return err
+		}
+		writeBufferId := utils.GetSHA256Hash(fileContents)
+		w, err := zipWriter.Create(writeBufferId)
+		if err != nil {
+			return err
+		}
+
+		if _, err := io.Copy(w, file); err != nil {
+			return err
+		}
+
+		if err = file.Close(); err != nil {
+			return err
+		}
+
+		// Remove the temporary file now that we've copied it into the zip file.
+		if err = os.Remove(filePath); err != nil {
+			return err
+		}
```

Comment:
I suggest doing this with a utility func outside of this method, it means that the contents of the input array are being destroyed as it progresses - this is an unexpected side-effect.

It would be better to do this all at once outside this function (e.g. by removing the dir holding the files)